diesel/src/mysql/connection/bind.rs

```diff
@@ -40,6 +71,56 @@ impl Binds {
         &mut self.mysql_binds
     }
 
+    /// The caller of this function must immediately check for errors
+    /// after return
+    pub fn populate_dynamic_buffers(&mut self, stmt: *mut ffi::MYSQL_STMT) {
+        use std::cmp::min;
+
+        for (i, buffer) in self.data.iter_mut().enumerate() {
+            if self.errors.as_ref().unwrap()[i] != 0 {
```

Comment:
Errors is none if this is an input bind, some if it's an output bind. This method makes no sense in the context of an output bind. Honestly, I should probably just separate these into two structs. There's basically no shared code other than that one function to link up the pointers.